### DISCUSSION TITLE
应用的某个版本正在使用已弃用的 Go 版本。请在 2019 年 10 月 1 日之前升级

### DISCUSSION THREAD

**Discussion by cixi666** [SeaHOH/GotoX#117] [一般]
https://cloud.google.com/appengine/docs/standard/go112/go-differences
https://cloud.google.com/appengine/docs/standard/go/sdk-gcloud-migration
是不是Go1.9在10月1日后不能用了？

**Comment by SeaHOH** [comment]
谢谢提醒，我还真没注意到。
不过下次请直接发链接，不要粘贴内容。

**Comment by cixi666** [comment]
好的

**Comment by SeaHOH** [comment]
发现个问题：现在除了 Python 2.7、Java 8、PHP 5，部署其它语言都需要开通结算账户，而这三个语言再过几年也可能被弃用。

暂时只能换回 Python 2.7 了，已经部署的 Go 1.9 应用好像不受影响，直到明年七月底。

**Comment by SeaHOH** [comment]
谷歌似乎未按计划停止接受 Go 1.9 应用的部署作业，不论新旧 SDK，现在都仍然可以正常使用此功能。

**Comment by SeaHOH** [comment]
关于新服务端计划请查看下方链接，非常欢迎提供各种意见和思路。
当然，也不一定非使用 GAE 不可，如果你觉得有哪些更好的选择 (仅限免费平台)，请告知并附上你认为其好的理由。
https://github.com/SeaHOH/GotoX/projects
https://github.com/SeaHOH/GotoX/projects/1

**Comment by k79e** [comment]
heroku免费的.....
其他人也过来分享下类似平台!!

**Comment by SeaHOH** [comment]
旧版本已无法部署

**Comment by k79e** [comment]
我听说旧版本在这个月某日都不给用了 也不清楚是不是真的. 月光他说的!!

**Comment by k79e** [comment]
我收到了一堆 都删了 看起来都是gcp的破广告也没注意看

邪了门了这评论怎么不按日期走了

**Comment by SeaHOH** [comment]
> 我听说旧版本在这个月某日都不给用了 也不清楚是不是真的. 月光他说的!!

倒是有可能。这是威廉的消息？哪里讲的，推上吗？
又要干活了。。不过真不想折腾啊，哪儿能来个大神把墙干翻，再也立不起来。。。

**Comment by k79e** [comment]
https://twitter.com/williamlong/status/1178873801515421697
看走眼了.. 写的是无法发布应用.

**Comment by SeaHOH** [comment]
为什么我没有收到过这个邮件，你呢，收到过没有？威廉应该不会乱讲吧。。。

**Comment by SeaHOH** [comment]
**Google App Engine Standard will use Google Cloud Build after November 30, 2019**
**书伴**也有收到过，但只我看到这个标题，没有原文。很奇怪为什么不发给所有用户。

看来只能先搞 heroku 了。

**Comment by SeaHOH** [comment]
> 邪了门了这评论怎么不按日期走了

github 的时间消失了一小时，现在正常了。。。

**Comment by Quantum2Bits** [comment]
Cloudflare 的 worker 是否可以考虑？ IBM 提供免费的docker.

**Comment by SeaHOH** [comment]
> Cloudflare 的 worker 是否可以考虑？

看了下文档，好像只能使用提供的 API，比如 Fetch，但是没看见限制条件。
还有 Free 方案的前 10M 请求是否就是指 10M 个外部 HTTP 请求，没看见详细说明。

> IBM 提供免费的docker.

这具体是什么意思？有现成的服务端 APP，只要写对应的客户端就好？

---

这几天我会先在 demo 上尝试下，如果方便的话，请也帮忙找一下说明各种限制条件的页面。

**Comment by k79e** [comment]
worker他们写的是读取次数 写的一点也不明白
> Unlimited (100k+) reads per second per key  (worker kv)

> 开发者帮助里面写的是日访问限制 Daily Request Limit

不清楚kv是存储盘还是什么东西(kv就是他们的存储盘 KV is a global, key-value data store that supports exceptionally high-read volumes with low-latency.)

我发现他的helloworld 请求一次那个控制面板请求数量就增加一次!!

**Comment by k79e** [comment]
ibm的就不给用 kube和openshift都不给免费的用
试用的话给送些金额但是要信用卡验证 那和google没什么区别了
我看还是cloudflare好 这个永久免费还省事

> Kubernetes Cluster 不可用于当前帐户类型。
所以他的免费是验证信用卡了之后才给解锁免费服务

Hyper Protect Virtual Server这个更坑人 验证之后给用 而且免费一律30日之后自动删掉实例 就是只给租30日.

**Comment by SeaHOH** [comment]
找到限制说明了 https://developers.cloudflare.com/workers/about/limits/

CFWorker 确实是个不错的选择，就是不知道能撑多久。

**Comment by SeaHOH** [comment]
对了，CFWorker 好像也不匿名，自己使用的时候需要记得这点。

**Comment by k79e** [comment]
CF财大气粗 哈哈.

**Comment by k79e** [comment]
我发现gopxy正好客户端也是go的 服务端是js的!!! 
不过这个东西bug严重 根本没法用. 真可惜哇.

**Comment by SeaHOH** [comment]
年底空闲少，我尽量抽空争取这个月内完成 CFWorker 支持，反正目标有了，不算太着急。

如果还发现有其它合适的平台，请告知，也可以加入支持。

**Comment by k79e** [comment]
接到通知 老应用当于0531后关闭
[Reminder] Google App Engine Standard apps still running on Go versions 1.9 and earlier will be shut down May 31, 2020
我发现worker其实多年前就有了 我当时看见都不知道他干啥的. 所以这个财大气粗的厂家一定能免费到底. 反正他就开蜜罐起家的 :D 所以这个东西一定是最可靠的之一.


> To continue running your App Engine Standard applications, redeploy these apps using Go 1.11 or above before May 31, 2020:
> If your applications use the App Engine Go SDK APIs, upgrade to Go 1.11.
> If you are not using these APIs, upgrade to Go 1.12.

**Comment by SeaHOH** [comment]
过几天就加上 CFWorker server，拖不少时间了。

**Comment by SeaHOH** [comment]
测试版 #142 已发布，今天熬夜总算是弄出来了。

不过 CFWorkers 的局限还是挺大的，说是支持 websocket，但测试暂时没有成功。10 万/日的请求数量一般使用估计问题不大，但还是会再试试能不能利用不限文件大小的特性搞个通道，用一个请求负载多个代理请求。还有就是 IP 数量的问题，解析结果只有 IPv4 和 IPv6 各 2 个，不知道能不能多搞点可用的 IP？

等 CFWorkers 完善点了，再搞下一个，欢迎大家继续推荐新的免费服务。我已经相中一个目标 KubeSail，就是不知道它能不能持续提供免费服务。

**Comment by k79e** [comment]
https://community.cloudflare.com/t/websocket-pass-through-crashes-worker-script/78482/6
cf他们说只要保留请求头就能让ws流量通过的样子.

**Comment by k79e** [comment]
KubeSail不支持私有镜像的样子 那怎么用啊? 还是后面可以支持了?
2020-02-19 14:12:25 看起来现在好像支持了.

**Comment by SeaHOH** [comment]
看了你提供的信息我好像明白了，原始请求的 body stream read 部分被我转到 fetch 请求的 body，然后就不能再被 Woker 转发，看来需要单独写个 API，我再试试。

**Comment by k79e** [comment]
openshift对某些应用 有封锁机制
检测机制不明 有时候开了引用没过多久就封了 有时候.....过几天突然被封了.
有人知道openshift对什么地方敏感么? 
看起来不像是匹配文件hash那样封的

kubesail还真不错 就是每7日要点击1下续命url 否则就关停应用

**Comment by k79e** [comment]
无服务器平台
https://www.netlify.com/

**Comment by SeaHOH** [comment]
> 无服务器平台
> https://www.netlify.com/

现在很忙，有空再看。

**Comment by Quantum2Bits** [comment]
GAE今天不能用了。

**Comment by Quantum2Bits** [comment]
GAE 服务器6月底失效，会更新或者支持XX-Net的服务器吗？

**Comment by SeaHOH** [comment]
GAE 代理不再维护，想使用 XX-Net 的服务器直接用它的客户端不就好了。

**Comment by Quantum2Bits** [comment]
XX-Net 不能绑定域名，我们这校园网有问题。

**Comment by SeaHOH** [comment]
你指的是 GAE 代理？当前客户端仍然支持 GAE 代理，但这需要用户自行重新部署服务端。过段时间，如果没人使用我会移除支持以提高效率。

**Comment by Quantum2Bits** [comment]
现在GotoX的GAE是唯一可以绑定域名的，现在还能用。已经收到Google的邮件说6月底不能用Go 1.9以前的版本。
[Reminder] Google App Engine Standard apps still running on Go versions 1.9 and earlier will be shut down June 30, 2020
有点担心6月底之后不能用GAE了。

**Comment by SeaHOH** [comment]
已经说得很明白了，绑卡用 Go 的新版本，我不愿意绑卡，所以要继续用的人请自己动手部署。

**Comment by k79e** [comment]
发现几个平台
https://anvil.works/ py的
https://www.pythonanywhere.com/
https://vercel.com/ 无服务器应用
https://gigalixir.com/ 不知道是啥

**Comment by SeaHOH** [comment]
谢谢。不过短时间内不会再添加更多服务端支持。
当前程序的框架很糟糕，扩展功能比较麻烦。也考虑过完全重构个更好的，但实在是没有那个精力，自身水平也有限，不一定做个新的就能完全解决问题，所以一直拖着。 :trollface:
